packages/ai-jsx/src/batteries/constrained-output.tsx

```diff
@@ -257,3 +268,46 @@ async function* ObjectCompletionWithRetry(
     }
   );
 }
+
+/**
+ * A {@link ChatCompletion} component that constrains the output to be a valid JSON string.
+ * It (ab)uses OpenAI function calls to generate the JSON string.
+ *
+ * @returns A string that is a valid JSON or throws an error.
+ */
+export async function JsonChatCompletionFunctionCall(
```

Comment:
Is this intended to be used publicly? Or are people supposed to use `JsonChatCompletion`, which delegates to this if possible?

Reply:
The idea was to delegate. But open to suggestions.

Reply:
On that note, yes we should probably remove the `export` keyword.

Reply:
And, if it's internal only, I was expecting to see a `@hidden` tag, to prevent it from showing up in the generated API docs.